resourceid_test.go

```diff
@@ -0,0 +1,38 @@
+package flux
+
+import (
+	"testing"
+)
+
+func TestResourceIDParsing(t *testing.T) {
+	type test struct {
+		name, id string
+	}
+	valid := []test{
+		{"full", "namespace:kind/name"},
+		{"legacy", "namespace/service"},
+		{"dots", "namespace:kind/name.with.dots"},
+		{"colons", "namespace:kind/name:with:colons"},
+		{"punctuation in general", "name-space:ki_nd/punc_tu:a.tion-rules"},
```

Comment:
You can actually have more than one `:`. In this case it's not specifying the namespace and name on opposite sides of it. This is a pattern that seems to only happen with clusterroles (from what I've seen). 

```
ubectl get clusterroles | grep ':'
system:aggregate-to-admin                                              1d
system:aggregate-to-edit                                               1d
system:aggregate-to-view                                               1d
system:auth-delegator                                                  1d
system:aws-cloud-provider                                              1d
system:basic-user                                                      1d
system:certificates.k8s.io:certificatesigningrequests:nodeclient       1d
system:certificates.k8s.io:certificatesigningrequests:selfnodeclient   1d
system:controller:attachdetach-controller                              1d
system:controller:certificate-controller                               1d
system:controller:clusterrole-aggregation-controller                   1d
system:controller:cronjob-controller                                   1d
system:controller:daemon-set-controller                                1d
<trimed>
```

Reply:
For example it leads to invalid names for deployments and other objects
```
k apply -f nginx.yaml
The Deployment "flux:nginx" is invalid: metadata.name: Invalid value: "flux:nginx": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.'
```

Reply:
Not sure what you're saying here. What's the content of nginx.yaml and where did it come from?

Reply:
Also: how did those cluster roles get those names, if kubernetes (well, kubectl) rejects names like that?

Reply:
The nginx.yaml just had a name set that was `flux:nginx` but was a very basic nginx deployment with nothing else in it. I'm saying the `:` is only allowed for roles and role bindings. Both namespaces and at the cluster level. Not for all resources. But also that it can have numerous `:` in it. The `system` ones above are standard on a new K8s cluster spun up with kubeadm.

Reply:
That works for me. Just need to update to allow multiple colons.

Reply:
The second to last test case above has two colons in the name.